Assets/HoloToolkit/Input/Scripts/Utilities/CameraMotionInfo.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using HoloToolkit.Unity.InputModule;
+using HoloToolkit.Unity;
+
+public class CameraMotionInfo : Singleton<CameraMotionInfo> {
+
+    #region Public accessors
+    public Vector3 HeadVelocity { get { return headVelocity; } }
+    public Vector3 MoveDirection { get { return headMoveDirection; } }
+
+    public float HeadZoneSizeIdle = 0.2f;
+    public float HeadZoneSizeMin = 0.01f;
+    #endregion
+
+    #region Private members
+    private Vector3 headVelocity;
+    private Vector3 lastHeadPos;
+    private Vector3 lastHeadZone;
+    private Vector3 newHeadMoveDirection;
+    private float headZoneSize = 1f;
+    private Vector3 headMoveDirection = Vector3.one;
+    #endregion
+
+    /// <summary>
+    /// Sends raycast from Main camera and returns focusable object
+    /// </summary>
+    public RaycastHit RayCastFromHead()
+    {
+        Transform head = Camera.main.transform;
+
+        RaycastHit hitInfo;
+
+        Ray gazeRay = new Ray(head.position, head.forward);
+
+        if (Physics.Raycast(gazeRay, out hitInfo))
```

Comment:
Why are we raycasting again?  Couldn't we get the current ray from the GazeManager?